ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -828,31 +827,19 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager, reconcileConcu
 }
 
 func (r *RayClusterReconciler) updateStatus(instance *rayiov1alpha1.RayCluster) error {
+	// TODO (kevin85421): ObservedGeneration should be used to determine whether update this CR or not.
+	instance.Status.ObservedGeneration = instance.ObjectMeta.Generation
+
 	runtimePods := corev1.PodList{}
 	filterLabels := client.MatchingLabels{common.RayClusterLabelKey: instance.Name}
 	if err := r.List(context.TODO(), &runtimePods, client.InNamespace(instance.Namespace), filterLabels); err != nil {
 		return err
 	}
 
-	count := utils.CalculateAvailableReplicas(runtimePods)
-	if instance.Status.AvailableWorkerReplicas != count {
-		instance.Status.AvailableWorkerReplicas = count
-	}
-
-	count = utils.CalculateDesiredReplicas(instance)
-	if instance.Status.DesiredWorkerReplicas != count {
-		instance.Status.DesiredWorkerReplicas = count
-	}
-
-	count = utils.CalculateMinReplicas(instance)
-	if instance.Status.MinWorkerReplicas != count {
-		instance.Status.MinWorkerReplicas = count
-	}
-
-	count = utils.CalculateMaxReplicas(instance)
-	if instance.Status.MaxWorkerReplicas != count {
-		instance.Status.MaxWorkerReplicas = count
-	}
+	instance.Status.AvailableWorkerReplicas = utils.CalculateAvailableReplicas(runtimePods)
```

Comment:
Is this a logic change, or is it purely syntax?

Reply:
The behaviors are functionally identical. This is merely a minor refactoring.

Reply:
Sounds good!